### PR TITLE
Set up Playroom

### PR DESCRIPTION
**Pull request by colebemis**
## TL;DR

I set up [Playroom](https://github.com/seek-oss/playroom) for Primer Components. 

> Playroom allows you to create a zero-install code-oriented design environment, built into a standalone bundle that can be deployed alongside your existing design system documentation.

 👉 Try it out: https://primer-components-d5t2omj9p.now.sh/playroom

## Problem

**We don’t have an easy way to build and share simple prototypes using Primer Components.**

Currently, the best ways to experiment with Primer Components are [live code examples](https://primer.style/doctocat/usage/live-code) and [CodeSandbox](https://codesandbox.io/).

Live code examples are a great way to make documentation more interactive. However, they're not designed for building out non-trivial prototypes. The editing interface is limiting and you can't save or share edited code examples.

CodeSandbox on the otherhand is great for creating complex prototypes. However, for simple prototypes, it’s a bit overkill. It’s a fully-featured IDE in the browser which can sometimes be overwhelming.

On a spectrum of simple to complex prototyping tools, live code examples belong on the left and CodeSandbox belongs on the right. We're missing something in the middle.

image

## Solution

This is where Playroom comes in. Playroom is a nice middle ground between live code examples and CodeSandbox.

image

Here's what it looks like:

primer-components-d5t2omj9p now sh_playroom_

With Playroom, we can:
- Iterate on designs in the final medium.
- Create quick mock-ups and interactive prototypes with real code.
- Exercise and evaluate the flexibility of our design system.
- Share work with others by simply copying the URL.

Playroom also supports autocomplete (powered by TypeScript), snippets, and multiple viewports.

### Implementation 

Here are the steps I took to set up Playroom:

1. Installed `playroom` as a devDependency (`yarn add -D playroom`)
2. Created the following files to configure Playroom: 
  * [`./playroom.config.js`](https://github.com/primer/components/pull/690/files#diff-bc8d3e7e12e0a3939eb0a98328256209) 
  * [`./playroom/FrameComponent.js`](https://github.com/primer/components/pull/690/files#diff-f866f47f0b9aa8226933e32ea740be30) (This is a component used to wrap all the previews with the `BaseStyles` component)
  * [`./playroom/snippets.js`](https://github.com/primer/components/pull/690/files#diff-a783357862e0401651d9f87088b4c4ed)
3. Added npm scripts to develop and build the playroom site:
  * `playroom:start` – Start a local development server (localhost:9000)
  * `playroom:build` – Build assets for production
  * `build` – Now automatically looks for this `build` script to generate static assets for the site
4. Updated `now.json` to deploy Playroom to `/playroom`

## Impact

People will be able to go to `primer.style/playroom` to build and share prototypes using Primer Components.

## Open questions

- Is `primer.style/playroom` the right URL?
- Should we link to Playroom from the docs? If so, where in the docs should we link to it?

## Merge checklist
- [ ] ~~Added or updated TypeScript definitions (`index.d.ts`) if necessary~~
- [ ] ~~Added/updated tests~~
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

File: now.json [hunk 1]
```diff
@@ -2,15 +2,9 @@
...
-    ".babelrc",
-    "package-lock.json",
-    "docs",
-    "rollup.config.js",
-    "src",
-    "static"
-  ],
+  "files": [".babelrc", "package-lock.json", "docs", "rollup.config.js", "src", "static"],
   "routes": [
+    {"src": "/playroom(/.*)?", "dest": "$1"},
```
Comment (emplums) [inline now.json]:
```suggestion
    {"src": "/components(/.*)?", "dest": "/playground$1"},
```

Comment (colebemis) [reply now.json]: This doesn't  work 😢  `primer.style/components/playroom` was my original plan but I've tried all sorts of things and can't get it to work. I agree that `/components/playroom` makes more sense but it might not be worth the hassle right now. Would you be okay with just `/playroom` for now, until we can get `/components/playroom` working?

Comment (BinaryMuse) [reply now.json]: Should this be `/components/playroom`? `/components` is already mapped on the line below

Comment (colebemis) [reply now.json]: ^ I tried that. No dice

Comment (colebemis) [reply now.json]: I figured `{"src": "/components/playroom(/.*)?", "dest": "$1"}` was our best bet. However, with that setup, the page 404s when loading all the JavaScript files because it's sending requests to `localhost:9000/components/runtime.js` instead of `localhost:9000/runtime.js`.

Comment (emplums) [reply now.json]: I'm fine with using `/playroom` for now!

Comment (emplums) [reply now.json]: It looks like line 8 404's anyways, not sure if it's even being used anymore

**Comment by emplums** [comment]
Would you be up to adding a new page in the side nav under `Getting Started` that explains what Playroom is, how to use it & link out the the playground with a large button or something easy to find?

**Comment by colebemis** [comment]
Sure! Since the `primer.style/playroom` link won't work until I update the now.json in http://github.com/primer/primer.style, I think it's best to tackle that in a couple PRs:
- PR 1: Set up playroom (that's this PR)
- PR 2: Add `/playroom` redirect in `primer/primer.style`
- PR 3: Add a new page to the Primer Components docs explaining Playroom

Does that work for you, @emplums?

**Comment by emplums** [comment]
@colebemis that plan looks good to me! 🙌
